File: CSharp/Solver1.cs
namespace ProjectEuler
{
    class Solver1 : ISolver
    {
        public int Solve()
        {
            return Calculate(1, 1000);
        }

        static int Calculate(int i, int limit)
        {
            return (i >= limit) ? 0 : ((i % 3 == 0 || i % 5 == 0) ? i : 0) + Calculate(i + 1, limit);
        }
    }
}

File: C/Solver1.c
int calculate1(int i, int limit)
{
    return (i >= limit) ? 0 : ((i % 3 == 0 || i % 5 == 0) ? i : 0) + calculate1(i + 1, limit);
}

int Solver1()
{
	return calculate1(1, 1000);
}
        